contracts/proxy_contract/v1/ergoscript/phoenix_v1_hodlcoin_proxy.es

```diff
@@ -185,39 +184,48 @@
         }
 
     } else {
-        
-        // ===== Refund Tx ===== //
-        val validRefundTx Boolean = {
 
-            // Outputs
-            val buyerPKBoxOUT: Box = OUTPUTS(0)
-            val minerFeeBoxOUT: Box = OUTPUTS(1)
+        // ===== Refund Tx ===== //
+        val validRefundTx: Boolean = {
+
+            val recommendedMinerFee: Long = 1000000L // overrides buyer's miner fee choice since they could go too low or too high
+
+            //ensures buyer receives total value of box
+            val validValueTransfer: Boolean = OUTPUTS.map { (o: Box) =>
+                if (o.propositionBytes == buyerPK.propBytes) o.value else 0L
+            }.fold(0L, { (a: Long, b: Long) => a + b }) >= SELF.value
```

Comment:
You need to do SELF.value - recommendedMinerFee

Reply:
these loops are unnecessary

Reply:
Miner fee needs to come as an input. Please see [this](https://github.com/PhoenixErgo/phoenix/blob/106680cde07eeb80038f92d708af78dcc0d2b0eb/src/test/scala/mockchain/ProxySpec.scala#L1015)

Scenario is that someone may need to refund hodl coins. If they sent only hodl coins + minErg then the box won't have enough to pay for a refund, or it may become negative which will break everything.